source/base/function_spherical.cc

```diff
@@ -225,7 +225,7 @@ namespace Functions
   template <int dim>
   SymmetricTensor<2,dim>
   Spherical<dim>::hessian (const Point<dim> &p_,
-                           const unsigned int component) const
+                           const unsigned int /* component */) const
```

Comment:
@bangerth no, i don't ;-) this is a scalar-valued function, so `component` is unused.